<launch>
  <arg name="robot_ip" default="192.168.125.1"/>
  <arg name="rws_username" default="ROS"/>
  <arg name="rws_password" default="robotics"/>
  <arg name="load_robot_description" default="true"/>
  <arg name="moveit_config_pkg" default="abb_irb120_moveit_config"/>
  <arg name="debug" default="false"/> <arg name="info" default="true"/> 

  <arg name="use_moveit" default="true"/>

  <!-- ############################## -->
  <!-- ### Robot Hardware Bringup ### -->
  <!-- ############################## -->

  <!-- Launch the bringup -->
  <include file="$(find com_3d)/launch/bringup_irb120.launch">
    <arg name="robot_ip" value="$(arg robot_ip)"/>
    <arg name="rws_username" value="$(arg rws_username)"/>
    <arg name="rws_password" value="$(arg rws_password)"/>
    <arg name="debug" value="$(arg debug)"/>
  </include>

  <!-- Spawn the robot and rsp -->
  <!-- <include file="$(find com_3d)/launch/load_irb120_3_58.launch" /> -->
  <param name="robot_description" command="$(find xacro)/xacro '$(find abb_irb120_gazebo)/urdf/irb120_3_58.xacro'" />
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="rsp" output="screen"/>

  <!-- ############################### -->
  <!-- ## Startup Script (EGM init) ## -->
  <!-- ############################### -->

  <!-- TEMP start rapid, etc. Let's see if we can do this in ros-syntax instead of a script -->
  <node pkg="com_3d"
        type="startup_robot.sh"
        name="robot_startup_automation"
        output="screen"
        launch-prefix="bash -c 'sleep 5.0; $0 $@' "/>

  <!-- ############################## -->
  <!-- ####### MoveIt Bringup ####### -->
  <!-- ############################## -->

  <!-- Load the moveit controllers -->
  <rosparam if="$(arg use_moveit)" command="load" file="$(find com_3d)/config/moveit_controllers.yaml"/>
  
  <!-- Planning context -->
  <include if="$(arg use_moveit)" file="$(find abb_irb120_moveit_config)/launch/planning_context.launch">
    <arg name="load_robot_description" value="false"/>
  </include>

  <include if="$(arg use_moveit)" file="$(find abb_irb120_moveit_config)/launch/move_group.launch">
    <arg name="publish_monitored_planning_scene" value="true"/>
    <arg name="allow_trajectory_execution" value="true"/>
    <arg name="fake_execution" value="false"/> 
    <arg name="info" value="$(arg info)"/>
    <arg name="debug" value="$(arg debug)"/>
  </include>

  <!-- MoveIt RViz visualization -->
  <include if="$(arg use_moveit)" file="$(find abb_irb120_moveit_config)/launch/moveit_rviz.launch">
    <arg name="config" value="true"/>
  </include>

  <!-- ############################### -->
  <!-- ### Video & AprilTag Config ### -->
  <!-- ############################### -->

  <!-- USB camera node -->
  <node pkg="usb_cam" type="usb_cam_node" name="usb_cam" output="screen">
    <param name="video_device" value="/dev/video0"/>
    <param name="image_width" value="1280"/>
    <param name="image_height" value="720"/>
    <param name="pixel_format" value="mjpeg"/>  <!-- or “yuyv”, depending on your camera -->
    <param name="io_method" value="mmap"/>
    <param name="frame_rate" value="30"/>
  </node>

  <!-- AprilTag detector (use the provided launch) -->
  <include file="$(find apriltag_ros)/launch/continuous_detection.launch">
    <!-- Remap camera topics -->
    <arg name="camera_name"  value="usb_cam"/>
    <arg name="image_topic"  value="/usb_cam/image_raw"/>
    <!-- <arg name="camera_info_topic" value="/usb_cam/camera_info"/> -->

    <!-- Family & tag config -->
    <!-- <arg name="tag_family"   value="tag36h11"/> -->
    <!-- <arg name="publish_tf"   value="false"/> -->
    <!-- You can also supply a tags.yaml via the 'tag_family'/'tag_samplers' args in that launch.
         If not, detections still work; metric translation needs correct size. -->
  </include>

  

</launch>